File: C++/shutdown.cpp
#include <iostream>
#include <thread>
#include <chrono>
#include <cstdlib>
using namespace std;

#if __linux__
    const string shutdown = "shutdown -h +";
    const string cancel = "shutdown -c";
    const int multiply = 1;
#elif _WIN32
    const string shutdown = "shutdown /s /t ";
    const string cancel = "shutdown /a";
    const int multiply = 60;
#endif

int main(){    
    int menu, choice;
    while(true){
        system("cls||clear");
        cout<<"1- Set the shutdown time\n2- Cancel the shutdown\n3- Exit\n\n>>>";
        cin>>menu;
        system("cls||clear");
        if (menu==1){
            cout<<"How many minutes later do you want the computer to shutdown: ";
            cin>>choice;
            system((shutdown+to_string(multiply*choice)).c_str());
        }
        else if (menu==2){
            system(cancel.c_str());
            cout<<"The shutdown has been canceled.."<<endl;
            std::this_thread::sleep_for(std::chrono::seconds(2));
        }
        else if (menu==3){
            break;
        }
        else{
            cout<<"Please make a valid choice.."<<endl;
            this_thread::sleep_for(std::chrono::seconds(2));
        }
    }
    return 0;
}

File: Rust/shutdown.rs
use std::time::Duration;
use std::thread::sleep;
use std::io::{self, Write};
use std::process::{Command, exit};

fn main() {
    let mut shell = "sh";
    let mut arg1 = "-c";
    let mut arg2 = "shutdown -h +";
    let mut cancel = "shutdown -c";
    let mut clear = "clear";
    let mut time = 1;
    if cfg!(windows) {
        shell = "cmd";
        arg1 = "/C";
        arg2 = "shutdown -s -f -t ";
        cancel = "shutdown -a";
        clear = "cls";
        time = 60;
    } 

    loop {
        Command::new(shell).arg(arg1).arg(clear).status().unwrap();
        println!("1- Set the shutdown time");
        println!("2- Cancel the shutdown");
        println!("3- Exit");
        print!("\n>>>");
        io::stdout().flush().unwrap();
        let mut menu = String::new();
        io::stdin().read_line(&mut menu).expect("Error!");
        let menu: u32 = match menu.trim().parse::<u32>() {
            Ok(num) => num,
            Err(_) => continue
        };

        Command::new(shell).arg(arg1).arg(clear).status().unwrap();

        match menu {
            1 => {
                print!("How many minutes later do you want the computer to shutdown: ");
                io::stdout().flush().unwrap();
                let mut minute = String::new();
                io::stdin().read_line(&mut minute).expect("Error!");
                let minute: u32 = match minute.trim().parse::<u32>() {
                    Ok(num) => num,
                    Err(_) => continue,
                };
                let minute = time*minute;
                
                let output = Command::new(shell).arg(arg1).arg(format!("{}{}", arg2,minute)).output().expect("Error!");
                if let Ok(stderr) = String::from_utf8(output.stderr){
                    Command::new(shell).arg(arg1).arg(clear).status().unwrap();
                    println!("{}", stderr);
                    sleep(Duration::from_secs(3));
                }
            }
            2 => {
                Command::new(shell).arg(arg1).arg(cancel).status().expect("Error!");
                println!("The shutdown has been canceled..");
                sleep(Duration::from_secs(1));
            }
            3 => exit(0),
            _ => {
                println!("Please make a valid choice..");
                sleep(Duration::from_secs(1));
            }
        }
    }
}